app/bench_test.go

```diff
@@ -36,14 +38,26 @@ func BenchmarkERC20Transfer(b *testing.B) {
 		require.NoError(b, err)
 		benchmarkERC20Transfer(b, db)
 	})
+	b.Run("memiavl", func(b *testing.B) {
+		benchmarkERC20Transfer(b, nil)
+	})
 }
 
+// pass `nil` to db to use memiavl
 func benchmarkERC20Transfer(b *testing.B, db dbm.DB) {
 	txsPerBlock := 1000
 	gasPrice := big.NewInt(100000000000)
 
+	var appOpts servertypes.AppOptions = EmptyAppOptions{}
+	if db == nil {
+		appOpts = AppOptionsMap(map[string]interface{}{
+			memiavlstore.FlagMemIAVL: true,
+		})
+	}
+
 	encodingConfig := MakeEncodingConfig()
-	app := New(log.NewNopLogger(), db, nil, true, true, map[int64]bool{}, DefaultNodeHome, 0, encodingConfig, EmptyAppOptions{})
+	app := New(log.NewNopLogger(), db, nil, true, true, map[int64]bool{}, b.TempDir(), 0, encodingConfig, appOpts)
+	defer app.Close()
```

Comment:
maybe can add a close for db if not nil

Reply:
yeah, but the `db` is passed from outside, normally it's who create it should close it?

Reply:
yea, but anyway it'll get closed after test